source/src/mesh/cs_MeshModelMulticastAcked.cpp

```diff
@@ -374,47 +374,53 @@ void MeshModelMulticastAcked::checkDone() {
 	if (_queueIndexInProgress == queue_index_none) {
 		return;
 	}
-	auto item = _queue[_queueIndexInProgress];
+	auto& item = _queue[_queueIndexInProgress];
 
 	// Check acks.
 	if (_ackedStonesBitmask.isAllBitsSet()) {
 		LOGi("Received ack from all stones.");
 		printMeshQueueItem(" ", item.metaData);
 
-		// TODO: get cmd type from payload in case of CS_MESH_MODEL_TYPE_CTRL_CMD
-		CommandHandlerTypes cmdType = MeshUtil::getCtrlCmdType((cs_mesh_model_msg_type_t)item.metaData.type);
-
-		result_packet_header_t ackResult(cmdType, ERR_SUCCESS);
-		UartHandler::getInstance().writeMsg(UART_OPCODE_TX_MESH_ACK_ALL_RESULT, (uint8_t*)&ackResult, sizeof(ackResult));
-		LOGMeshModelDebug("all success");
+		CommandHandlerTypes cmdType = static_cast<CommandHandlerTypes>(item.controlCommand);
+		if (cmdType == CTRL_CMD_UNKNOWN) {
+			LOGMeshModelDebug("Control command is unknown: don't send ack result");
+		}
+		else {
+			result_packet_header_t ackResult(cmdType, ERR_SUCCESS);
+			LOGMeshModelInfo("Ack all result: commandType=%u returnCode=%u", ackResult.commandType, ackResult.returnCode);
+			UartHandler::getInstance().writeMsg(UART_OPCODE_TX_MESH_ACK_ALL_RESULT, (uint8_t*)&ackResult, sizeof(ackResult));
+		}
 
 		remQueueItem(_queueIndexInProgress);
 		_queueIndexInProgress = queue_index_none;
 	}
 
 	// Check for timeout.
 	if (_processCallsLeft == 0) {
```

Comment:
Rename _processCallsLeft to something like _FooTimeout to explain what happens on ==0?

Reply:
Didn't even touch this in this PR ;) Also, I think the comment in the header explains it well enough.